gamestonk_terminal/stocks/sector_industry_analysis/financedatabase_view.py

```diff
@@ -0,0 +1,280 @@
+        )
+    else:
+        print("No company found. No barchart will be depicted.")
+    print("")
+
+    export_data(
+        export,
+        os.path.dirname(os.path.abspath(__file__)),
+        finance_metric,
+        pd.DataFrame.from_dict(stocks_data),
+    )
+
+
+def display_companies_per_sector(country: str, mktcap: str = "", export: str = ""):
+    """
+    Display number of companies per sector in a specific country (and market cap). [Source: Finance Database]
+
+    Parameters
+    ----------
+    country: str
+        Select country to get number of companies by each sector
+    mktcap: str
+        Select market cap of companies to consider from Small, Mid and Large
+    export: str
+        Format to export data as
+    """
+    companies_per_sector = financedatabase_model.get_companies_per_sector(
+        country, mktcap
+    )
+
+    companies_per_sector = dict(
+        OrderedDict(
+            sorted(companies_per_sector.items(), key=lambda t: t[1], reverse=True)
+        )
+    )
+
+    legend, values = zip(*companies_per_sector.items())
+
+    colors = [
+        "b",
+        "g",
+        "r",
+        "c",
+        "m",
+        "y",
+        "k",
+        "tab:blue",
+        "tab:orange",
+        "tab:gray",
+        "lightcoral",
+        "yellow",
+        "saddlebrown",
+        "lightblue",
+        "olive",
+    ]
+
+    plt.subplots(figsize=plot_autoscale(), dpi=PLOT_DPI)
+    if gtff.USE_ION:
+        plt.ion()
+    plt.pie(
+        values,
+        labels=legend,
+        colors=colors,
+        wedgeprops={"linewidth": 0.5, "edgecolor": "white"},
+    )
+    plt.title(f"{mktcap + ' cap c' if mktcap else 'C'}ompanies per sector in {country}")
+    plt.tight_layout()
+
+    plt.show()
+
+    export_data(
+        export,
+        os.path.dirname(os.path.abspath(__file__)),
+        "cps",
+        pd.DataFrame.from_dict(companies_per_sector),
+    )
+
+
+def display_companies_per_industry(country: str, mktcap: str = "", export: str = ""):
+    """
+    Display number of companies per industry in a specific country. [Source: Finance Database]
+
+    Parameters
+    ----------
+    country: str
+        Select country to get number of companies by each industry
+    mktcap: str
+        Select market cap of companies to consider from Small, Mid and Large
+    export: str
+        Format to export data as
+    """
+    companies_per_industry = financedatabase_model.get_companies_per_industry(
+        country, mktcap
+    )
+
+    colors = list(mcolors.CSS4_COLORS.keys())[::6]
+
+    companies_per_industry = dict(
+        OrderedDict(
+            sorted(companies_per_industry.items(), key=lambda t: t[1], reverse=True)
+        )
+    )
+
+    # are there more industries than colors
+    if len(companies_per_industry) > len(colors):
+        companies_per_industry_sliced = dict(
+            list(companies_per_industry.items())[: len(colors) - 2]
```

Comment:
And it helps to not perform too many actions in one line of code. You are here making a sub selection of the length of colors (minus 2) of a dictionary converted into a list converted into a dictionary. It also makes me wonder if it wouldn't be tidier to use a Series in this case, then you still have the labels and can select like you are using lists.

All in all, I think I am looking for some sort of coding style standard (other than PEP8) for the project.

Reply:
The reason why I went for this design is because the only thing I needed was to slice a dictionary based on the value. This is why I convert the items to list, slice and reconvert back to dictionary. The variable didn't change, only got sliced, this is also why the name of the variable is `companies_per_industry_sliced`. Would it help having:
`companies_per_industry_sliced: Dict = ...`?

The use of colours for the slicing could actually be better, because I rely on the size of `mcolors.CSS4_COLORS` which may not be ideal in case it changes in the future. Will improve this.

Yup, PEP8 is the coding style standard we have been using. Apart from that the code has the same "convention" through black linters and through code reviews.